modules/appeals_api/app/services/appeals_api/pdf_construction/supplemental_claim/v3/form_data.rb

```diff
@@ -0,0 +1,191 @@
+            'loanGuaranty' => 7,
+            'lifeInsurance' => 8,
+            'nationalCemeteryAdministration' => 9
+          }.freeze
+
+          CLAIMANT_TYPE_CODES = {
+            'veteran' => 1,
+            'spouse_of_veteran' => 2,
+            'child_of_veteran' => 3,
+            'parent_of_veteran' => 4,
+            'other' => 5
+          }.freeze
+
+          LONG_SIGNATURE_THRESHOLD = 70
+          LONG_EMAIL_THRESHOLD = 120
+          MAX_SIGNATURE_LENGTH = 180
+
+          def initialize(supplemental_claim)
+            @supplemental_claim = supplemental_claim
+          end
+
+          delegate :veteran_dob_month, :veteran_dob_day, :veteran_dob_year, :signing_appellant_zip_code,
+                   :date_signed, :signing_appellant, :appellant_local_time, :contestable_issues,
+                   :new_evidence_locations, :claimant_type_other_text, :new_evidence_dates, :claimant, :veteran,
+                   :alternate_signer_full_name,
+                   to: :supplemental_claim
+
+          delegate :first_name, :last_name, :middle_initial, :file_number, :service_number,
+                   :insurance_policy_number,
+                   to: :veteran, prefix: true
+
+          delegate :first_name, :last_name, :middle_initial,
+                   to: :claimant, prefix: true
+
+          delegate :email, :number_and_street, :city, :state_code, :country_code,
+                   to: :signing_appellant, prefix: true
+
+          def benefit_type_code
+            BENEFIT_TYPE_CODES[supplemental_claim.benefit_type]
+          end
+
+          def claimant_type_code
+            CLAIMANT_TYPE_CODES[supplemental_claim.claimant_type]
+          end
+
+          def form_5103_notice_acknowledged
+            return nil unless supplemental_claim.benefit_type == 'compensation'
+
+            supplemental_claim.form_5103_notice_acknowledged ? 1 : 'Off' # 1 => 'YES' OFF => 'NO'
+          end
+
+          def new_evidence_locations
+            evidence_records.map(&:location)
+          end
+
+          def new_evidence_dates
+            evidence_records.map(&:dates)
+          end
+
+          def veteran_ssn_first_three
+            veteran.ssn[0..2]
+          end
+
+          def veteran_ssn_middle_two
+            veteran.ssn[3..4]
+          end
+
+          def veteran_ssn_last_four
+            veteran.ssn[5..8]
+          end
+
+          def signing_appellant_zip_code
+            if signing_appellant.zip_code_5 == '00000'
+              signing_appellant.international_postal_code || '00000'
+            else
+              signing_appellant.zip_code_5
+            end
+          end
+
+          def domestic_phone?
+            # The form has no field for an extension on a domestic number, so if a domestic number has
+            # an extension, we put it in the international field instead.
+            signing_appellant.domestic_phone? && signing_appellant.phone_data['phoneNumberExt'].blank?
+          end
+
```

Comment:
These data getters for phones and signatures return nil when they should not be used, even if data is technically available - I found that this helped avoid putting some logic in the form structure code (see my older PR linked in the description)